Plugin/PluginRegistry.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using BepInEx;
+using Silksong.ModMenu.Elements;
+using Silksong.ModMenu.Internal;
+using Silksong.ModMenu.Models;
+using Silksong.ModMenu.Screens;
+
+namespace Silksong.ModMenu.Plugin;
+
+public delegate bool PluginHandler(
+    IModMenuInterface plugin,
+    [MaybeNullWhen(false)] out SelectableElement menuElement
+);
+
+/// <summary>
+/// Registry of all IModMenuInterface extensions, used for generating mod menus from plugins.
+/// </summary>
+public static class PluginRegistry
+{
+    private static readonly List<PluginHandler> registeredHandlers =
+    [
+        HandleType<IModMenuCustomElement>(GenerateCustomElement),
+        HandleType<IModMenuCustomMenu>(GenerateCustomMenu),
+        HandleType<IModMenuToggle>(GenerateToggle),
+    ];
+
+    /// <summary>
+    /// Register a custom plugin interface handler.
+    /// </summary>
+    public static void AddHandler(PluginHandler handler) => registeredHandlers.Add(handler);
```

Comment:
Is it worth explaining how you'd use this a bit more - is the idea that an extension library could define a custom IModMenuInterface type, and then register a handler that client mods could use? I'm struggling to understand what this offers that can't be done with IModMenuCustomElement.